### PR TITLE
Integrate bal persist code generation to the `bal build` command

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose

This is the implementation of the proposal https://github.com/ballerina-platform/ballerina-library/issues/5784

## Checklist
- [x] Linked to an issue
- [x] Updated the specification
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: examples/build.gradle [hunk 1]
```diff
@@ -164,9 +164,9 @@ task initInMemoryExamples {
                     exec {
                         workingDir "${project.projectDir}/build/generated-examples/${example}"
                         if (!Os.isFamily(Os.FAMILY_WINDOWS)) {
-                            commandLine 'sh', '-c', "${ballerinaDist}/bin/bal persist init --module=entities"
+                            commandLine 'sh', '-c', "${ballerinaDist}/bin/bal persist generate --module=entities --datastore=inmemory"
```
Comment (daneshk) [inline examples/build.gradle:167]: Can we make `datastore` default to `inmemory` and `module` to root package?

File: persist-cli-tests/src/test/java/io/ballerina/persist/tools/utils/GeneratedSourcesTestUtils.java [hunk 2]
```diff
@@ -52,7 +54,7 @@ public class GeneratedSourcesTestUtils {
      * Represents persist commands.
      */
     public enum Command {
-        INIT,
+        ADD,
```
Comment (daneshk) [inline persist-cli-tests/src/test/java/io/ballerina/persist/tools/utils/GeneratedSourcesTestUtils.java:57]: Are we removing the INIT command completely?

File: examples/build.gradle (hunk 1 above)
Comment (Bhashinee) [reply examples/build.gradle:167]: As per this [comment](https://github.com/ballerina-platform/ballerina-library/issues/5784#issuecomment-1844989176) in the proposal, the `datastore` is a mandatory parameter for the new `generate` command. Have I misunderstood it?
The example is written with the import `import foo/medical_center.entities;` for using the generated client code which is inside the `entities` folder. So I kept the same. Should I change this?

File: persist-cli-tests/src/test/java/io/ballerina/persist/tools/utils/GeneratedSourcesTestUtils.java (hunk 2 above)
Comment (Bhashinee) [reply persist-cli-tests/src/test/java/io/ballerina/persist/tools/utils/GeneratedSourcesTestUtils.java:57]: No, we haven't removed the `INIT` command. It's just I converted all the `init` tests to the `add` command tests. I will add a test case for `init` as well.

File: docs/spec/spec.md [hunk 3]
```diff
@@ -130,12 +146,22 @@ The database schema will contain the code to create,
...
+|    --datastore    |  used to indicate the preferred database client. Currently, 'mysql', 'mssql', 'google sheets' and 'postgresql' are supported.   |    Yes     |        |
+|     --module      |      used to indicate the persist enabled module in which the files are generated.       |    No     | <package_name> |
+
 Behaviour of the `generate` command,
 - User should invoke the command within a Ballerina project
-- The user should have initiated the persistence layer in the project and update the model definition file.
 - The model definition file should contain the `persist` module import (`import ballerina/persist as _;`)
 - The Model definition file should contain at least one entity
-- If the user invokes the command twice, it will not fail. It will generate the files once again.
+- If the user invokes the command twice, it will fail. The user needs to remove the persist configurations from the Ballerina.toml if they want to reinitialize the project.
```
Comment (daneshk) [inline docs/spec/spec.md:164]: why can't we execute the generate command twice? which configuration is it?

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_27/Config.toml [hunk 4]
```diff
@@ -12,3 +12,4 @@ user = "root"
 password = ""
 database = ""
 
+
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_27/Config.toml:15]:
```suggestion
```

File: docs/spec/spec.md (hunk 3 above)
Comment (Bhashinee) [reply docs/spec/spec.md]: We can execute the generate command twice. Fixed in [5e25761](https://github.com/ballerina-platform/persist-tools/pull/331/commits/5e25761a1d5f197e29a863683df4b4ff22918950)

File: persist-cli/src/main/resources/version.properties [hunk 5]
```diff
@@ -1,4 +1,4 @@
 version = ${project.version}
-persistSqlVersion = ${project.persistSqlNativeVersion}
-persistInMemoryVersion = ${project.persistInMemoryNativeVersion}
-persistGoogleSheetsVersion = ${project.persistGoogleSheetsNativeVersion}
+persistSqlVersion = 1.2.2-SNAPSHOT
+persistInMemoryVersion = 1.2.0
+persistGoogleSheetsVersion = 1.2.0
```
Comment (daneshk) [inline persist-cli/src/main/resources/version.properties:4]: Don't change this file

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_init_1/persist/model.bal [hunk 6]
```diff
@@ -1,3 +1 @@
-import ballerina/persist as _;
-
-
+import ballerina/persist as _;
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_init_1/persist/model.bal:1]:
```suggestion
import ballerina/persist as _;

```

**Review by daneshk — DISMISSED** [review]
LGTM

**Review by daneshk — APPROVED** [review]

**Pull request merged**